backends/xnnpack/runtime/XNNPACKBackend.cpp

```diff
@@ -44,13 +44,17 @@ class XnnpackBackend final : public PyTorchBackendInterface {
         processed->size(),
         executor,
         context.get_runtime_allocator());
+    // This backend does not need its processed data after compiling the model.
+    processed->Free();
+
     if (err != Error::Ok) {
+      // destroy() won't be called on this handle, so we need to clean it up
+      // now.
+      executor->~XNNExecutor();
+
       ET_LOG(Error, "XNNCompiler::compleModel failed: 0x%x", (unsigned int)err);
+      return err;
```

Comment:
not gonna lie, I always wondered why when this failed we wouldn't immediately exit. But I never took the time to dig into it. Really nice catch. Thank you for this!